src/libraries/System.Private.CoreLib/src/System/IO/Strategies/DerivedFileStreamStrategy.cs

```diff
@@ -146,7 +146,11 @@ public override Task FlushAsync(CancellationToken cancellationToken)
         public override Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken)
             => _fileStream.BaseCopyToAsync(destination, bufferSize, cancellationToken);
 
-        public override ValueTask DisposeAsync() => _strategy.DisposeAsync();
+        public override async ValueTask DisposeAsync()
+        {
+            await _strategy.DisposeAsync().ConfigureAwait(false);
+            await _fileStream.BaseDisposeAsync().ConfigureAwait(false);
+        }
```

Comment:
I'm not sure what's the right Dispose pattern. Should we always call base first or last? 
If we call it first, I think we could also tackle this issue https://github.com/dotnet/runtime/issues/82186. Otherwise, the file handle will already be closed and we won't we be able to fix https://github.com/dotnet/runtime/issues/82186.

Reply:
@Jozkee how about just changing `FileStream.DisposeAsync` to:

```cs
        public async override ValueTask DisposeAsync()
        {
            await _strategy.DisposeAsync().ConfigureAwait(false);
            await base.DisposeAsync().ConfigureAwait(false);
        }
```

Reply:
> how about just changing FileStream.DisposeAsync to:

That won't fix it. We need `_fileStream.BaseDisposeAsync()` because that one calls the `Close()` of the type that extends FileStream. 

`base.DisposeAsync()` will call into the Close() method of DerivedFileStreamStrategy, which is not what is desired. 

There's also this comment hinting at what I just described: 
https://github.com/dotnet/runtime/blob/0a15c3b97dabaf94e5d4399aef2fd48c2b3f446a/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/DerivedFileStreamStrategy.cs#LL14C40-L14C40